basics/keeping-up-to-date/upgrade-module/upgrade-module-internal-behavior.md

```diff
@@ -80,6 +80,18 @@ Classes are gathered by responsibilities:
 - Classes interacting with PrestaShop core
 - Tasks (Can be considered as controllers for upgrade, rollback etc.)
 
+## Upgrade modules from local source
+
+During the autoupgrade process, in the "modules upgrade" step, we check for each module if there is a new version available in the PrestaShop Marketplace
+
+If there is one, we download it and then upgrade.
+
+That means, for developers, to test the autoupgrade process with your module, you must release it and push it to the Marketplace before.
+
+To ease the tests, you can put your module ZIP in `/ADMIN_DIR/autoupgrade/modules/MODULE_NAME.zip` and then use your local version to upgrade.
```

Comment:
```suggestion
Developers must release and push their module to the Marketplace in order to test the upgrade process, which is inconvenient. In order to simplify the testing process, you can place the ZIP file of your module in `/ADMIN_DIR/autoupgrade/modules/MODULE_NAME.zip`. This will allow you to use your local version for upgrading purposes.
```